rootex/framework/components/visual/model/sprite_component.cpp

```diff
@@ -0,0 +1,127 @@
+#include "sprite_component.h"
+
+#include "common/common.h"
+
+#include "event_manager.h"
+#include "core/resource_loader.h"
+#include "systems/render_system.h"
+
+DEFINE_COMPONENT(SpriteComponent);
+
+SpriteComponent::SpriteComponent(Entity& owner, const JSON::json& data)
+    : RenderableComponent(owner, data)
+    , m_ImageMaterialResourceFile(ResourceLoader::CreateBasicMaterialResourceFile("rootex/assets/materials/sprite.basic.rmat"))
+{
+	configureBoundingRectangle();
+}
+
+void SpriteComponent::configureBoundingRectangle()
+{
+	float m_ImageScaleFactor = 0.001;
+
+	Ref<Texture> imageTexture = m_ImageMaterialResourceFile->getDiffuseImageFile()->getTexture();
+	float rectWidth = imageTexture->getWidth() * m_ImageScaleFactor;
+	float rectHeight = imageTexture->getHeight() * m_ImageScaleFactor;
+
+	PRINT("Rect Width: " + std::to_string(rectWidth));
+	PRINT("Rect Height: " + std::to_string(rectHeight));
+
+	Vector<VertexData> vertices;
+	VertexData vertex;
+	ZeroMemory(&vertex, sizeof(VertexData));
+	for (int i = 0; i < 4; i++)
+	{
+		int u = (i == 1 || i == 2) ? 1 : 0;
+		int v = (i == 2 || i == 3) ? 1 : 0; 
+
+		vertex.position.x = u * rectWidth / 2;
+		vertex.position.y = v * rectHeight / 2;
+		vertex.position.z = 0;
+
+		vertex.normal.x = 0;
+		vertex.normal.y = 0;
+		vertex.normal.z = 1;
+
+		vertex.textureCoord.x = u;
+		vertex.textureCoord.y = v;
+
+		vertices.push_back(vertex);
+	}
+
+	Vector<unsigned short> indices = { 0, 1, 2, 2, 3, 0 };
+
+	m_VertexBuffer.reset(new VertexBuffer((const char*)vertices.data(), vertices.size(), sizeof(VertexData), D3D11_USAGE_IMMUTABLE, 0));
+	m_IndexBuffer.reset(new IndexBuffer(indices));
+}
+
+bool SpriteComponent::preRender(float deltaMilliseconds)
+{
+	ZoneNamedN(componentPreRender, "Sprite Pre-Render", true);
+	RenderableComponent::preRender(deltaMilliseconds);
+	return true;
+}
+
+void SpriteComponent::render(float viewDistance)
+{
+	ZoneNamedN(componentRender, "Sprite Render", true);
+	RenderableComponent::render(viewDistance);
+
+	RenderSystem::GetSingleton()->getRenderer()->bind(m_ImageMaterialResourceFile.get());
+	RenderSystem::GetSingleton()->getRenderer()->draw(m_VertexBuffer.get(), m_IndexBuffer.get());
```

Comment:
don't do `bind` calls in component, shift to material

Reply:
The actual bind calls are being done in the material only.

Reply:
my bad, missed the `Material` in `m_ImageMaterialResourceFile`